### ISSUE TITLE
Thingsboard service exits with status=137/n/a after upgrades

### ISSUE THREAD

**Issue by Dewieinns** [thingsboard/thingsboard#5648]
Hey all, I have an instance of Thingsboard CE running on a Digital Ocean droplet. It was working fine running 3.0.1 (slow - but it DID work) and last night I upgraded it all the way to 3.3.2 running through each version in between. The upgrade process worked as expected and Thingsboard starts and runs as expected initially, but it it crashes.

I have only 3 IoT devices connected to the instance, all of which send data every 5 - 10 minutes. 

The main service exits with the following message:
```
thingsboard.service: Main process exited, code=exited, status=137/n/a
/usr/share/thingsboard/bin/thingsboard.jar: line 278: 6750 Killed "$javaexe" "${arguments[@]}"
```

From what I can tell the status code means it "exited due to a OOM error", I'm running a droplet with 1GB and, again, this worked fine until I upgraded. Any documentation I can see says Thingsboard should still work with this limited amount of memory, and yes I have the config file edited to include:

`export JAVA_OPTS="$JAVA_OPTS -Xms256M -Xmx256M"`

There isn't any error messages in the /var/log/thingsboard/thingsboard.log, only an occasional few warnings were in the log:
```
[tb-queue-request-template-tb_transport.api.responses.ubuntu-Thingsboard-12-thread-1] WARN o.t.s.q.c.DefaultTbQueueRequestTemplate - Request timeout detected, currentNs [19670488688606], ResponseMetaData{submitTime=19636225877189, calculatedExpTime=19646225877189, deltaMs=10000000000, expTime=19646225877189, future=SettableFuture@2e812215[status=PENDING]}, key [fa52d40d-251b-41d3-8ecb-ead559b6da9e]

[HikariPool-1 housekeeper] WARN com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Thread starvation or clock leap detected (housekeeper delta=1m15s263ms474µs706ns).
```
Shortly after reporting the above to another medium I uninstalled Thingsboard completely and re-installed it restoring the DB data from a back-up and modifying the fresh new config file as necessary. Unfortunately the issue persisted. 

I then upgraded postgres and, again, the issue seemed to persist, particularly if I make a call for a large amount of data. 

When this happens I've verified postgres is still up and running. 

I have attached my logs file since the fresh install. 
[thingsboard.log](https://github.com/thingsboard/thingsboard/files/7626512/thingsboard.log)

**Comment by Dewieinns** [comment]
Well a bit of an update. I was speaking with a most special friend about my issue and he suggested I check the swap space on my server, turns out there was none. I have enabled a 2GB swap and I must say, the improvement in performance is incredible. 

I'm not sure if it's an actual fix yet as the system has only been running for about half an hour since I made the change but things that were crashing Thingsboard instantly (loading a lot of data) now seem to be working as expected (and with much better performance than before)
